server/app/repository/UserRepository.java

```diff
@@ -279,9 +280,9 @@ public void createNewApplicantForTrustedIntermediaryGroup(
     newAccount.save();
     Applicant applicant = new Applicant();
     applicant.setAccount(newAccount);
-    applicant
-        .getApplicantData()
-        .setUserName(form.getFirstName(), form.getMiddleName(), form.getLastName());
+    ApplicantData applicantData = applicant.getApplicantData();
+    applicantData.setUserName(form.getFirstName(), form.getMiddleName(), form.getLastName());
+    applicantData.setDOB(form.getDob());
```

Comment:
suggestion (blocking): validate the user-provided `dob` value in the form before writing it to the database, add tests to exercise the validation